waku/waku_archive/archive.nim

```diff
@@ -46,7 +46,7 @@ type DefaultMessageValidator* = ref object of MessageValidator
 
 method validate*(validator: DefaultMessageValidator, msg: WakuMessage): ValidationResult =
   if msg.timestamp == 0:
-    return ok()
+    return err(timestampIsZero)
```

Comment:
Although I agree with the REST API adding timestamps on behalf of the application when _publishing_ messages. I still don't think the Store should (for now) enforce sender timestamps. Doing so goes against the existing RFC. Furthermore, this obsoletes the entire `receiverTimestamp`, `storedAt`, etc. mechanism which is a larger cleanup exercise. We may indeed decide to follow this route (and perhaps should have from the beginning), but deprecating the RFC provision for optional timestamps is a bit larger, involves an RFC change and a more extensive review of the existing data scheme.